### PR TITLE
documents: create documents with item CLI

### PR DESCRIPTION
**Pull request by rerowep**
* Adds a cli for creation of documents with assoziated items.

Co-Authored-by: Peter Weber <peter.weber@rero.ch>

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* rero/rero-ils-ui#<xx>

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

**Review by zannkukai — APPROVED** [review]

File: rero_ils/modules/cli.py [hunk 1]
```diff
@@ -451,6 +455,217 @@ def create(infile, create_or_update, append, reindex, dbcommit, commit,
             )
 
 
+@utils.command('validate_documents_with_items')
+@click.argument('infile', type=click.File('r'), default=sys.stdin)
+@click.option('-v', '--verbose', 'verbose', is_flag=True, default=False)
+@click.option('-d', '--debug', 'debug', is_flag=True, default=False)
+@with_appcontext
+def validate_documents_with_items(infile, verbose, debug):
+    """Validates REROILS records with items.
```
Comment (iGormilhit) [inline rero_ils/modules/cli.py:464]:
```suggestion
    """Validate RERO ILS records with items.
```

File: rero_ils/modules/cli.py [hunk 2]
```diff
@@ -451,6 +455,217 @@ def create(infile, create_or_update, append, reindex, dbcommit, commit,
...
+
+    :param infile: Json file
+    :param verbose: verbose print
+    :param debug: print traceback
+    """
+    def add_org_lib_doc(item):
+        """Adds organisation, library and document to item for validation."""
```
Comment (iGormilhit) [inline rero_ils/modules/cli.py:471]:
```suggestion
        """Add organisation, library and document to item for validation."""
```

File: rero_ils/modules/cli.py [hunk 3]
```diff
@@ -451,6 +455,217 @@ def create(infile, create_or_update, append, reindex, dbcommit, commit,
...
+@click.option('-C', '--commit', 'commit', default=10000)
+@click.option('-v', '--verbose', 'verbose', is_flag=True, default=False)
+@click.option('-d', '--debug', 'debug', is_flag=True, default=False)
+@with_appcontext
+def create_documents_with_items(infile, lazy, dont_stop_on_error, save_errors,
+                                commit, verbose, debug):
+    """Load REROILS record with items.
+
+    :param infile: Json file
+    :param lazy: lazy reads file
```
Comment (iGormilhit) [inline rero_ils/modules/cli.py:555]:
```suggestion
    :param lazy: lazy read file
```

**Review by iGormilhit — CHANGES_REQUESTED** [review]
Commit message proposition:

```
documents: create documents with item CLI

* Extends item CLI for validating and creating documents with linked
  items.
```

**Review by iGormilhit — APPROVED** [review]

**Review by benerken — APPROVED** [review]

**Review by BadrAly — APPROVED** [review]

**Pull request merged**